src/v2/Apps/Conversation/Components/Conversation.tsx

```diff
@@ -175,6 +61,55 @@ const Conversation: React.FC<ConversationProps> = props => {
     />
   ))
 
+  // Pagination Scroll Logic
+  const [fetchingMore, setFetchingMore] = useState(false)
+  const [messagesTop, setMessagesTop] = useState(null)
+  const scrollContainer = useRef(null)
+
+  useEffect(() => {
+    if (!("IntersectionObserver" in window)) return
```

Comment:
**thought (non-blocking):**  I like the use of raw intersection observers here. Looking at [caniuse](https://caniuse.com/#feat=intersectionobserver), support is pretty broad (around ~92%) with the major holdout being IE11 (which we don't support anyway). In [other parts of the codebase](https://github.com/artsy/force/blob/4825ed8cf4f3861b39bccb26fbe934db1eaaa68e/src/desktop/apps/article/components/InfiniteScrollArticle.tsx#L91) we've used [react-waypoint](https://github.com/civiccc/react-waypoint) to accomplish a similar thing. It would be interesting to strike up a conversation in client-infra on if we should move towards more pure intersection observers or if there's a solid reason to stay on `react-waypoint`. 

<!--
Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.
-->

Reply:
Oh, I didn't know about `react-waypoint`. I assume that using the raw observer should be more performant, but I get what @damassi is saying in regards to `react-waypoint` already being in place in other parts, and keeping convention in how this is used. I'll refactor with `react-waypoint`. (I hope I'm understanding the combination of what was said on the channel and the thumbs up correctly 😬).

Reply:
There was no direct action requested here, I was just thinking out loud. This is a larger client infra question that would be good to figure out in the future. I'm good with the waypoint refactor though, because if we decide to trade waypoint out for something else we can just update that here too.